src/Symfony/Component/Mime/Tests/Part/TextPartTest.php

```diff
@@ -87,6 +88,35 @@ public function testEncoding()
         ), $p->getPreparedHeaders());
     }
 
+    public function testCustomEncoding()
+    {
+        $p = new TextPart('content', 'utf-8', 'plain', 'test_encoding');
+        $p->addEncoder('test_encoding', new class() implements ContentEncoderInterface {
+            public function encodeByteStream($stream, int $maxLineLength = 0): iterable
+            {
+                while (!feof($stream)) {
+                    yield fread($stream, 16372);
+                }
+            }
+
+            public function getName(): string
+            {
+                return 'test_encoding';
+            }
+
+            public function encodeString(string $string, ?string $charset = 'utf-8', int $firstLineOffset = 0, int $maxLineLength = 0): string
+            {
+                return $string;
```

Comment:
We should modify the given string in some form. Otherwise the test does not really show me that the method was actually called.